server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/methods/RequestHandler.java

```diff
@@ -76,18 +76,20 @@ public RequestHandler(Set<Method> methods, JmapRequestParser jmapRequestParser,
     private Function<Method, Flux<JmapResponse>> extractAndProcess(AuthenticatedRequest request) {
         MailboxSession mailboxSession = request.getMailboxSession();
         return (Method method) -> {
-                    try {
-                        JmapRequest jmapRequest = jmapRequestParser.extractJmapRequest(request, method.requestType());
-                        return method.process(jmapRequest, request.getMethodCallId(), mailboxSession)
-                            .onErrorResume(JmapFieldNotSupportedException.class, e -> errorNotImplemented(e, request));
-                    } catch (IOException e) {
-                        LOGGER.error("Error occured while parsing the request.", e);
-                        if (e.getCause() instanceof JmapFieldNotSupportedException) {
-                            return errorNotImplemented((JmapFieldNotSupportedException) e.getCause(), request);
-                        }
-                        return error(request, generateInvalidArgumentError(e.getMessage()));
-                    }
-                };
+            try {
+                JmapRequest jmapRequest = jmapRequestParser.extractJmapRequest(request, method.requestType());
+                return method.process(jmapRequest, request.getMethodCallId(), mailboxSession)
+                    .onErrorResume(JmapFieldNotSupportedException.class, e -> errorNotImplemented(e, request));
+            } catch (IOException e) {
+                LOGGER.error("Error occured while parsing the request.", e);
+                if (e.getCause() instanceof JmapFieldNotSupportedException) {
```

Comment:
Isn't `JmapFieldNotSupportedException` being caught below?

Reply:
We should wrap it's instanciation in a mono to simplify error handling here, for sure.